Assignment5/gork4mezua-21A307/assignment5.py

```diff
@@ -0,0 +1,355 @@
+      ?actor_id rdfs:label ?actor_label .
+      ?movie_id rdfs:label ?movie_label .
+      FILTER(?actor_id != wd:Q2685) .
+      FILTER(LANG(?actor_label) = 'en')
+      FILTER(LANG(?movie_label) = 'en')
+  }
+  ORDER BY ?actor_label
+''')
+sparql.setReturnFormat(JSON)
+results = sparql.query().convert()
+
+# Result (SPARQL)
+results_df = pd.io.json.json_normalize(results['results']['bindings'])
+results_df[['actor_label.value', 'movie_label.value']].head()
+
+# In Kypher:
+kgtk("""
+    query -i all \
+     --match '(movies)-[:P161]->(:Q2685),
+              (movies)-[:P161]->(actors)' \
+     --return 'actors' 
+    / add-labels
+""")
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+sparql.setQuery('''
+  SELECT ?award_label ?award
+  WHERE {
+      wd:Q2685 wdt:P166 ?award . # Q2685 (Arnold Schwarzenegger)
+      ?award rdfs:label ?award_label .
+      FILTER(LANG(?award_label) = 'en')
+  }
+  ORDER BY ?award_label
+''')
+sparql.setReturnFormat(JSON)
+results = sparql.query().convert()
+
+# Result (SPARQL)
+results_df = pd.io.json.json_normalize(results['results']['bindings'])
+results_df[['award_label.value', 'award.value']].head()
+
+# In Kypher:
+kgtk("""
+    query -i all \
+     --match '(:Q2685)-[:P166]->(awards)' \
+     --return 'awards' 
+    / add-labels
+""")
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+sparql.setQuery('''
+  SELECT ?member_label ?member_id ?political_label ?political_id 
+  WHERE {
+      wd:Q2685 wdt:P102 ?political_id .
+      ?member_id wdt:P102 ?political_id .
+      ?member_id wdt:P31 wd:Q5 .
+      ?member_id rdfs:label ?member_label .
+      ?political_id rdfs:label ?political_label .
+      FILTER(LANG(?member_label) = 'en') .
+      FILTER(LANG(?political_label) = 'en')
+  }
+  ORDER BY ?member_label
+  LIMIT 5
+''')
+sparql.setReturnFormat(JSON)
+results = sparql.query().convert()
+
+# Result (SPARQL)
+results_df = pd.io.json.json_normalize(results['results']['bindings'])
+results_df[['member_label.value', 'political_label.value']].head()
+
+# In Kypher:
+kgtk("""
+    query -i all \
+     --match '(:Q2685)-[:P102]->(parties),
+              (members)-[:P102]->(parties),
+              (members)-[:P31]->(:Q5)' \
+     --return 'members' 
+    / add-labels
+""")
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL:
+sparql.setQuery('''
+  SELECT DISTINCT ?prop ?val
+  WHERE {
+      wd:Q483501 ?prop ?val .
+      ?wdprop wikibase:directClaim ?prop .
+      ?wdprop wdt:P31+ wd:Q107649491
+  }
+''')
+sparql.setReturnFormat(JSON)
+results = sparql.query().convert()
+
+# Result (SPARQL)
+results_df = pd.io.json.json_normalize(results['results']['bindings'])
+results_df[['prop.value', 'val.value']].head()
+
+# In Kypher:
+kgtk("""
+    query -i all \
+     --match '(class)-[:P279*]->(:Q483501),
+              (class)-[l {label: property}]->(value)' \
```

Comment:
This is not correct. You are printing the properties of the class instead of the elements (P106)